pypeit/pypeit_steps.py

```diff
@@ -0,0 +1,914 @@
+
+    sci_ID and det need to have been set internally prior to calling this method
+
+    Args:
+        spectrograph (:class:`~pypeit.spectrographs.spectrograph.Spectrograph`):
+            The spectrograph instance
+        fitstbl (:class:`~pypeit.metadata.PypeItMetaData`):
+            The class holding the metadata for all the frames in this PypeIt run.
+        par (:class:`~pypeit.par.pypeitpar.PypeItPar`):
+            The parameter set for the reduction, including slitmask and
+            object finding parameters.
+        frames (:obj:`list`):
+            List of frames to extract; stacked if more than one
+            is provided
+        det (:obj:`int`):
+            Detector number (1-indexed)
+        calib_ID (:obj:`str`):
+            The calibration group ID
+        calibrations_path (:obj:`str`):
+            Path to the calibration files
+        sciImg (:class:`~pypeit.images.pypeitimage.PypeItImage`):
+            Data container that holds a single image from a
+            single detector and its related images (e.g. ivar, mask)
+        bkg_redux_sciimg (:class:`~pypeit.images.pypeitimage.PypeItImage`, optional):
+            Data container that holds a single image from a
+            single detector and its related images (e.g. ivar, mask)
+            before background subtraction if self.bkg_redux is True,
+            otherwise None. It's used to generate a global sky
+            model without bkg subtraction.
+        initial_sky (`numpy.ndarray`_):
+            Initial global sky model
+        sobjs_obj (:class:`~pypeit.specobjs.SpecObjs`):
+            List of objects found during `run_objfind`
+        bkg_redux (:obj:`bool`, optional):
+            Indicates whether the reduction involves background subtraction.
+            Default is False.
+        find_negative (:obj:`bool`, optional):
+            Indicates whether to find negative objects during the reduction.
+            Default is False.
+        calib_slits (:class:`~pypeit.slittrace.SlitTraceSet`, optional):
+            If provided, use these slits instead of those from the
+            calibrations. Default is None.
+        show (:obj:`bool`, optional):
+            Show the QA during processing. Default is False.0
+
+    Returns:
+        tuple: A tuple containing:
+            - spec2DObj (:class:`~pypeit.spec2dobj.Spec2DObj`):
+              The 2D spectrum object containing the 2D spectral images
+            - sobj (:class:`~pypeit.specobjs.SpecObjs`):
+                The extracted 1D spectra
+            
+    """
+    # Grab some meta-data needed for the reduction from the fitstbl
+    #self.objtype, self.setup, self.obstime, self.basename, self.binning \
+    #        = self.get_sci_metadata(frames[0], det)
+    objtype, setup, obstime, basename, binning \
+            = get_sci_metadata(spectrograph, fitstbl, frames[0], det)
+
+    # Grab the calibrations
+    caliBrate = load_calibrations_for_frame(
```

Comment:
@profxj I found a problem here, while I was testing. Since the calibrations are loaded from file here, the bpm slit flagging happening during obj finding is not propagated here. This causes inconsistency between the information form `slits` and what is saved in `specobj` from objfind and cause the code to crash. I wonder if we need to save to file also an intermediate slittrace object? This problem I think will happen also if we want to use the maskdesign information in a step-by-step reduction.